uaclient/files/notices.py

```diff
@@ -0,0 +1,153 @@
+        label="contract_refresh_warning", order_id="40"
+    )
+    DAEMON_AUTO_ATTACH_LOCK_HELD = NoticeFileDetails(
+        label="daemon_auto_attach_lock_held", order_id="50"
+    )
+    DAEMON_AUTO_ATTACH_FAILED = NoticeFileDetails(
+        label="daemon_auto_attach_failed", order_id="51"
+    )
+    OPERATION_IN_PROGRESS = NoticeFileDetails(
+        label="operation_in_progress", order_id="60"
+    )
+
+
+class Notices:
+    def __init__(
+        self,
+        is_permanent: bool = True,
+        root_mode: bool = True,
+        override_dir: Optional[str] = None,
+    ):
+        if not override_dir:
+            self.directory = "/var/lib/ubuntu-advantage/notices/"
+            if not is_permanent:
+                self.directory = "/var/run/notices/"
+            self.override = False
+        else:
+            self.directory = override_dir
+            self.override = True
+        self.is_root = root_mode
+        self.is_permanent = is_permanent
+
+    def add(self, notice_label_id: NoticeFileDetails, description: str):
+        label = notice_label_id.label
+        file_order_id = notice_label_id.order_id
+        if self.is_root:
+            notice_files = self.get_all_notice_files()
+            labels_present = [file.split("-")[-1] for file in notice_files]
+            if label in labels_present:
+                return
+            filename = "{}-{}".format(file_order_id, label)
+            system.write_file(
+                os.path.join(self.directory, filename), description
+            )
+        else:
+            raise exceptions.NonRootUserError
+
+    def try_add(self, notice_label_id: NoticeFileDetails, description: str):
+        try:
+            self.add(notice_label_id, description)
+        except exceptions.NonRootUserError:
+            event.warning("Trying to add notice as non-root user")
+
+    def remove(self, notice_label_id: NoticeFileDetails):
+        label = notice_label_id.label
+        file_order_id = notice_label_id.order_id
+        filename = "{}-{}".format(file_order_id, label)
+        if self.is_root:
+            notice_files = self.get_all_notice_files()
+            label_file = [file for file in notice_files if file == filename]
+            if len(label_file) == 1:
+                system.remove_file(os.path.join(self.directory, label_file[0]))
+        else:
+            raise exceptions.NonRootUserError
+
+    def try_remove(self, notice_label_id: NoticeFileDetails):
+        try:
+            self.remove(notice_label_id)
+        except exceptions.NonRootUserError:
+            event.warning("Trying to add notice as non-root user")
+
+    def get_all_notice_files(self):
+        if not os.path.exists(self.directory):
+            os.makedirs(self.directory)
+
+        notices = [
+            file
+            for file in os.listdir(self.directory)
+            if os.path.isfile(os.path.join(self.directory, file))
+        ]
+        notices.sort()
```

Comment:
I think we should sort based on the priority of the notice here right ?  I know this will require us to match the file title with existing notices, but maybe it is worth it.

Reply:
The file names are lexicographically made, the restart notices are highest in the priority. We can discuss the other notices priority.

Reply:
Right, because we name the files with the priority number first, we can just sort them by file name :+1: